src/Voron/Data/Tables/TableSchema.IndexDefinitions.cs

```diff
@@ -0,0 +1,443 @@
+                }
+
+                Debug.Assert(GenerateKey.Method.DeclaringType?.FullName != null && GenerateKey.Method.DeclaringType.Assembly.FullName != null,
+                    $"Invalid {nameof(GenerateKey)} '{GenerateKey.Method.Name}'");
+
+                var assemblyName = new AssemblyName(GenerateKey.Method.DeclaringType.Assembly.FullName);
+                var declaringType = $"{GenerateKey.Method.DeclaringType.FullName}, {assemblyName.Name}";
+                var declaringTypeBytes = Encodings.Utf8.GetBytes(declaringType);
+                fixed (byte* ptr = declaringTypeBytes)
+                {
+                    serializer.Add(ptr, declaringTypeBytes.Length);
+                }
+
+                byte[] serialized = new byte[serializer.Size];
+
+                fixed (byte* destination = serialized)
+                {
+                    serializer.CopyTo(destination);
+                }
+
+                return serialized;
+            }
+
+            public static DynamicKeyIndexDef ReadFrom(ByteStringContext context, byte* location, int size)
+            {
+                var input = new TableValueReader(location, size);
+                var indexDef = new DynamicKeyIndexDef();
+
+                byte* currentPtr = input.Read(1, out _);
+                indexDef.IsGlobal = Convert.ToBoolean(*currentPtr);
+
+                currentPtr = input.Read(2, out var currentSize);
+                Slice.From(context, currentPtr, currentSize, ByteStringType.Immutable, out indexDef.Name);
+
+                // read IndexEntryKeyGenerator method name
+                currentPtr = input.Read(3, out currentSize);
+                var methodName = Encodings.Utf8.GetString(currentPtr, currentSize);
+
+                // read IndexEntryKeyGenerator declaring type
+                currentPtr = input.Read(4, out currentSize);
+                var declaringType = Encodings.Utf8.GetString(currentPtr, currentSize);
+
+                //var type = System.Type.GetType(declaringType);
+                var type = System.Type.GetType(declaringType);
+
+                Debug.Assert(type != null, $"Invalid data, failed to get {nameof(GenerateKey)}.Method.DeclaringType from deserialized value : {declaringType}");
+
+                var method = type.GetMethod(methodName, BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Instance | BindingFlags.Static);
+
+                Debug.Assert(method != null, $"Invalid data, failed to get method-info from type : {type}, method name : {methodName}");
+                Debug.Assert(method.IsStatic, $"Invalid data, {nameof(GenerateKey)} must be a static method. method name : {methodName}");
+                Debug.Assert(method.GetCustomAttribute<IndexEntryKeyGeneratorAttribute>() != null, 
+                    $"Invalid data, {nameof(GenerateKey)} must be marked with custom attribute '{nameof(IndexEntryKeyGeneratorAttribute)}'. method name : {methodName}");
+
```

Comment:
Those need to be _regular checks_. The reason we have those is to prevent arbitrary code execution, so we need that to run even in release mode

Reply:
done